activitypub-core/src/main/java/org/xwiki/contrib/activitypub/internal/json/ActivityPubObjectDeserializer.java

```diff
@@ -51,38 +55,45 @@
 
     /**
      * {@inheritDoc}
+     * <p>
      * Deserialize the current object based on the given "type" property. If the type attribute does not exist, or  if
      * no class is found with the same type name, then the object is deserialized using {@link ActivityPubObject} as
      * fallback.
      */
     @Override
     public ActivityPubObject deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
-        throws IOException, JsonProcessingException
+        throws IOException
     {
         ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
         ObjectNode root = mapper.readTree(jsonParser);
 
+        Logger logger = (Logger) deserializationContext.findInjectableValue(LOGGER_KEY, null, null);
+
         JsonNode type = root.get("type");
         Class<? extends ActivityPubObject> instanceClass;
         if (type != null) {
-            instanceClass = findClass(type.asText());
+            instanceClass = findClass(type.asText(), logger).orElse(UnknownTypeObject.class);
         } else {
-            instanceClass = ActivityPubObject.class;
+            instanceClass = UnknownTypeObject.class;
```

Comment:
I would perform a bit more work here, to put the actual ObjectNode inside the UnknownTypeObject, or even its serialization. By doing so, it will allow us: 
  1. to debug more easily in case of unknown type by seeing where it cames from exactly, what it's supposed to represent
  2. maybe later to properly serialize the object with all the information, that could be very handy on the long term to ensure we store the right information. We can imagine that for now some information will be stored even if we don't know to handle some types, while in the future we might be able to handle them, so better not loose info in storage.

Reply:
This is a good idea, I have create https://jira.xwiki.org/browse/XAP-102 to implement this in the next release.